ldap/servers/slapd/connection.c

```diff
@@ -963,57 +1069,81 @@ connection_free_private_buffer(Connection *conn)
 #define CONN_TURBO_HYSTERESIS 0          /* avoid flip flopping in and out of turbo mode */
 
 void
-connection_make_new_pb(Slapi_PBlock *pb, Connection *conn)
+connection_set_new_op_in_pb(Slapi_PBlock *pb, pc_tinfo_t *tinfo, Connection *conn)
 {
-    struct Slapi_op_stack *stack_obj = NULL;
-    /* we used to malloc/free the pb for each operation - now, just use a local stack pb
-     * in connection_threadmain, and just clear it out
+    /*
+     * Init the pblock with conn and a fresh operation
+     * Operation no more realloced or got from a stack but
+     * it is stored in worker thread context and reused afterwards
+     * Caller should held conn->c_mutex
      */
-    /* *ppb = (Slapi_PBlock *) slapi_ch_calloc( 1, sizeof(Slapi_PBlock) ); */
-    /* *ppb = slapi_pblock_new(); */
+    Slapi_Operation *op = tinfo->op;
```

Comment:
tinfo->op is set when g_pc_do_not_reuse_operation() is called. It looks surprising to call operation_init without testing 'op'

Reply:
No problem here because operation_init checks for NULL operation